test/Sentry.AspNetCore.Tests/SentryTracingMiddlewareTests.cs

```diff
@@ -316,7 +316,7 @@ public async Task Baggage_header_propagates_to_outbound_requests(bool shouldProp
                 "sentry-trace_id=75302ac48a024bde9a3b3734a82e36c8, " +
                 "sentry-public_key=d4d82fc1c2c4032a83f3a29aa3a3aff, " +
                 "sentry-sample_rand=0.1234, " +
-                "sentry-sample_rate=0.5";
+                "sentry-sample_rate=1";
```

Comment:
Why isn't this propagating the result of the incoming baggage header anymore?

Reply:
Spoke to @Flash0ver about this... there's another scenario which is that SDK users can force `IsSampled` via the `ITransactionContext` when calling `StartTransaction`... and once we implement the backpressure management, it might get overwritten by downsampling. So looks like we always have to run this logic.

Reply:
update: see https://github.com/getsentry/sentry-dotnet/pull/4374#discussion_r2235738488